### PR TITLE
Orders Co-mentions by number of "mentions"

### PR DESCRIPTION
**Pull request by Zachary-Dernar**
Adds unit test for sorting results by mention count.

### CODE REVIEW

File: test/views/MentionsAndSentiments/test.index.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import assert from 'assert';
+import { shallow } from 'enzyme';
+import Accordion from '../../../views/Accordion/index.jsx';
+import MentionsAndSentiments from '../../../views/MentionsAndSentiments/index.jsx';
+
+describe('foo', () => {
```
Comment (maniax89) [inline test/views/MentionsAndSentiments/test.index.js]: how bout a better name this this?

File: test/views/MentionsAndSentiments/test.index.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  let mentionsAAA = {
```
Comment (maniax89) [inline test/views/MentionsAndSentiments/test.index.js]: i don't know what `AAA` is

File: views/MentionsAndSentiments/index.jsx [hunk 3]
```diff
@@ -40,7 +39,12 @@ export default React.createClass({
             negative: item[2],
             neutral: item[3],
           })),
-      }));
+    }));
+
+    mentions = mentions.sort(function(a, b) {
```
Comment (maniax89) [inline views/MentionsAndSentiments/index.jsx]: you could write this line as
`mentions = mentions.sort((a, b) => return (b.data.length - a.data.length))`

File: test/views/MentionsAndSentiments/test.index.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      to: '20170501'
+    },
+    text: 'Sample Company'
+  };
+
+  it('Orders results by number of "mentions"', () => {
+
+    let wrapper = shallow(<MentionsAndSentiments query={queryAAA} mentions={mentionsAAA} />);
+    let firstMention = wrapper.find(Accordion).nodes[0].props.header.props.children[0].props.children;
+    let secondMention = wrapper.find(Accordion).nodes[1].props.header.props.children[0].props.children;
```
Comment (besh) [inline test/views/MentionsAndSentiments/test.index.js]: These definitely can't be looked up via class name? Like, if you do `wrapper.find(Accordion).debug()` do you see class name attrs anywhere in the logged tree?

File: views/MentionsAndSentiments/index.jsx (hunk 3 above)
Comment (besh) [reply views/MentionsAndSentiments/index.jsx]: I think you guys should add a lint rule for anonymous arrow functions.

File: test/views/MentionsAndSentiments/test.index.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import assert from 'assert';
+import { shallow } from 'enzyme';
+import Accordion from '../../../views/Accordion/index.jsx';
+import MentionsAndSentiments from '../../../views/MentionsAndSentiments/index.jsx';
```
Comment (besh) [inline test/views/MentionsAndSentiments/test.index.js:5]: Something for a future PR, I think you'll want to either alias directories, resolve imports starting with the app entry point, or maybe include tests along side of the actual component. This pattern is going to be brutal to maintain.

File: test/views/MentionsAndSentiments/test.index.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  let queryAAA = {
+    date: {
+      from: '20170301',
+      to: '20170501'
+    },
+    text: 'Sample Company'
+  };
+
+  it('Orders results by number of "mentions"', () => {
+
```
Comment (besh) [inline test/views/MentionsAndSentiments/test.index.js]: Extra line here.

File: test/views/MentionsAndSentiments/test.index.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+          },
+        ]
+      }
+    ]
+  };
+
+  let queryAAA = {
+    date: {
+      from: '20170301',
+      to: '20170501'
```
Comment (besh) [inline test/views/MentionsAndSentiments/test.index.js]: You don't have quotes around the keys here and have a space between the key/value but do not follow this convention with the `mentionsAAA` object. I think this convention here is better but the main thing is to be consistent. A good case for a couple lint rules.

File: package.json [hunk 8]
```diff
@@ -8,7 +8,7 @@
     "test-integration": "casperjs test ./test/integration/test.*.js",
     "test-integration-runner": "NODE_ENV=test node casper-runner.js",
     "test": "npm run lint && npm run test-unit && npm run test-integration-runner",
-    "test-unit": "istanbul cover ./node_modules/mocha/bin/_mocha test/unit",
+    "test-unit": "./node_modules/mocha/bin/_mocha --compilers js:babel-core/register test/views/MentionsAndSentiments test/unit",
```
Comment (maniax89) [inline package.json]: does this work if we just do `./node_modules/mocha/bin/_mocha test/views test/unit`? then create a file `./test/mocha.opts` with the `--compilers js:babel-core/register` option? see http://mochajs.org/#mochaopts

File: package.json [hunk 9]
```diff
@@ -61,14 +61,17 @@
     "whatwg-fetch": "^2.0.1"
   },
   "devDependencies": {
+    "babel-core": "^6.24.1",
     "casperjs": "^1.1.3",
     "codecov": "^1.0.1",
```
Comment (maniax89) [inline package.json]: we can remove `codecov`

File: package.json [hunk 10]
```diff
@@ -61,14 +61,17 @@
...
+    "enzyme": "^2.8.2",
     "eslint": "^3.13.1",
     "eslint-plugin-import": "^2.2.0",
     "eslint-plugin-react": "^6.9.0",
     "istanbul": "^0.4.5",
```
Comment (maniax89) [inline package.json]: we can remove `istanbul`

File: test/views/MentionsAndSentiments/test.index.js [hunk 11]
```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import assert from 'assert';
+import { shallow } from 'enzyme';
+import Accordion from '../../../views/Accordion/index.jsx';
+import MentionsAndSentiments from '../../../views/MentionsAndSentiments/index.jsx';
+
+describe('mentions_sort_test', () => {
```
Comment (maniax89) [inline test/views/MentionsAndSentiments/test.index.js]: could it be called `describe('<MentionsAndSentiments />', () => {` ? that way if we decide to add more tests later we don't have to rename it

File: views/MentionsAndSentiments/index.jsx [hunk 12]
```diff
@@ -40,7 +39,10 @@ export default React.createClass({
             negative: item[2],
             neutral: item[3],
           })),
-      }));
+    }));
+
+    mentions = mentions.sort((a, b) => (b.data.length - a.data.length))
+    .filter((name, i) => i < 4);
```
Comment (maniax89) [inline views/MentionsAndSentiments/index.jsx]: nitpick - would add 2 spaces here so its indented slightly

File: test/views/MentionsAndSentiments/test.index.js (hunk 11 above)
Comment (besh) [reply test/views/MentionsAndSentiments/test.index.js]: I really dig the `<ComponentName />` describes.

**Review by maniax89 — APPROVED** [review]

**Comment by besh** [comment]
LGTM.

**Pull request merged**